packages/pwa-kit-dev/src/utils/resolver-utils.js

```diff
@@ -0,0 +1,132 @@
+
+    // Sanitize the input. Here we want to remove the file extension and index file if it exists.
+    sourcePath = sourcePath.split('.')[0]
+    sourcePath = sourcePath.split(path.sep).join('/')
+    sourcePath = sourcePath.replace(indexRegExp, '')
+
+    // Do the same for the import path even thought it's not common to use /index and file extensions in your module
+    // imports.
+    importPath = importPath.split('.')[0]
+    importPath = importPath.replace(indexRegExp, '')
+
+    return sourcePath.endsWith(importPath)
+}
+
+/**
+ * Normalize and expand the extension configuration array so that it is easier to process.
+ * @param {{String, Object}[]} extensions - The extensions configuration value as defined in the PWA-Kit config.
+ * @returns {Object[]} extensions - The extensions array in object form.
+ *
+ * @example
+ * const result = expand(["store-finder", ["account-pages", {singlePage: true}], './extensions/local-extension']);
+ * console.log(result)
+ * // [["@salesforce/extension-store-finder", {}], ["@salesforce/extension-account-pages", {singlePage: true}], ["/home/project/extensions/local-extension", {}]]
+ */
+export const expand = (extensions = []) =>
+    extensions
+        .filter((extension) => Boolean(extension))
+        .map((extension) => {
+            let [nameRef, config = {}] = Array.isArray(extension) ? extension : [extension, {}]
+
+            switch (true) {
+                case /^\./.test(nameRef):
+                    // Relative Path
+                    nameRef = nameRef.split(/\/|\\/).join(path.sep)
+                    nameRef = path.join(process.cwd(), nameRef.replace('.', ''))
+                    break
+                case /^(\/|\\|\w:)/.test(nameRef):
+                    // Absolute Path (UNIX|Windows)
+                    nameRef = nameRef.split(/\/|\\/).join(path.sep)
+                    break
+                case /^@/.test(nameRef):
+                    // Do nothing
+                    break
+                default:
+                    // Default is to treat the reference as a extension "short" name.
+                    nameRef = `${EXTENSION_NAMESPACE}/${EXTENSION_PREFIX}-${nameRef}`
+                    break
+            }
+
+            return [nameRef, config]
+        })
+
+/**
+ * Based on the current extensibility configuration, return an array of candiate file paths to be used
+ * in the wild-card import module resolution for the given import path..
+ *
+ * @param {String} importPath - The import module-name.
+ * @param {String} sourcePath - The path the the file of the source import.
+ * @returns {String[]} paths - The potential paths to find the module import.
+ */
+export const buildCandidatePathArray = (importPath, sourcePath, opts = {}) => {
+    // Replace wildcard character as it has done its job getting us to this point.
+    importPath = importPath.replace('*/', '')
+
+    const {extensions = getConfig()?.app?.extensions} = opts
```

Comment:
nit 🤔 i wonder if we should make `extensions` a required argument for this function and so it no longer calls `getConfig` in the implementation. We can always get the config from the webpack plugin. This removes the pwa-kit-runtime dependency from this utility function and it make be easier to test and maintain?

Reply:
I think that might be a good idea. Give me some time to think about how it will all work in the bigger picture of using this utility with the resolver plugins for webpack and babel and I'll get back to you.